nodes/src/nodes/agent_crewai/crewai.py

```
@@ -285,20 +347,215 @@ def _on_any_event(source, event):
+        # Not thread-safe; safe because pipeline runs are sequential per node instance.
+        self._current_pSelf: Any = None
+
+    def run_agent(self, pSelf: Any, question: Any, *, emit_answers_lane: bool = True) -> Any:
+        """Override to stash pSelf before delegating to AgentBase.run_agent()."""
+        self._current_pSelf = pSelf
+        try:
+            return super().run_agent(pSelf, question, emit_answers_lane=emit_answers_lane)
+        finally:
+            self._current_pSelf = None
+
+    def _run(
```

Ruff (0.15.7)

[warning] Too many statements (53 > 50)

(PLR0915)